core/new-gui/src/app/workspace/component/navigation/navigation.component.ts

```diff
@@ -325,6 +326,39 @@ export class NavigationComponent implements OnInit {
     this.workflowActionService.deleteOperatorsAndLinks(allOperatorIDs, []);
   }
 
+  public onClickImportWorkflow = (file: NzUploadFile): boolean => {
+    const reader = new FileReader();
+    reader.readAsText(file as any);
+    reader.onload = () => {
+      const result = reader.result;
+      if (typeof result === "string") {
+        const workflow = JSON.parse(result) as Workflow;
+        this.workflowActionService.reloadWorkflow(workflow, true);
+      }
+    };
+    return false;
+  };
+
+  public onClickExportWorkflow(): void {
+    const workflow: Workflow = this.workflowActionService.getWorkflow();
+    // make a copy of the workflow with erased metadata info, such as workflow id
+    const workflowCopy: Workflow = {
+      ...workflow,
+      wid: undefined,
+      creationTime: undefined,
+      lastModifiedTime: undefined,
+    };
+
+    var workflowJson = JSON.stringify(workflowCopy);
+    var element = document.createElement("a");
```

Comment:
This is kind of tentative. You might want to refactor this.

Reply:
done